src/dotnet/Chat.UI.Blazor/Components/ContactSelect/ContactSelect.razor

```diff
@@ -20,13 +21,31 @@
 
 @code {
     private List<ChatInfo> _filteredChats = new();
-    private readonly List<ChatId> _selectedChats = new();
+    private ContactSelectContext.Model _model = null!;
 
     [Inject] private ChatListUI ChatListUI { get; init; } = null!;
+    [Inject] private IContacts Contacts { get; init; } = null!;
 
+    [CascadingParameter] public ContactSelectContext.Model? ContextModel { get; set; }
+
+    [Parameter] public bool ShowUserContactsOnly { get; set; }
+    [Parameter] public ChatListOrder ChatListOrder { get; set; } = ChatListOrder.ByOwnUpdateTime;
     [Parameter, EditorRequired] public SearchPhrase SearchPhrase { get; set; } = null!;
     [Parameter] public IReadOnlyCollection<ChatId> ContactsToExclude { get; set; } = Array.Empty<ChatId>();
     [Parameter] public EventCallback<IReadOnlyCollection<ChatId>> SelectionChanged { get; set; }
+    [Parameter] public EventCallback<IReadOnlyCollection<ChatInfo>> ChatSelectionChanged { get; set; }
+
+    protected override void OnInitialized() {
+        base.OnInitialized();
+        _model = ContextModel ?? new ContactSelectContext.Model();
+        _model.Changed += async (_, _) => {
```

Comment:
can event handler's code throw?

Reply:
I think it's responsibility of event handler to handle exception. We never catch exception of event handlers inside components.